source/kernel/src/drivers/serial_mux.rs

```diff
@@ -355,33 +345,34 @@ impl IncomingMuxerTask {
                 if ch.last() != Some(&0) {
                     // This is the last chunk, and it doesn't end with a zero.
                     // just add it to the accumulator, if we can.
-                    if (self.idx + ch.len()) <= self.buf.len() {
-                        self.buf[self.idx..][..ch.len()].copy_from_slice(ch);
-                        self.idx += ch.len();
-                    } else {
-                        warn!("Overfilled accumulator");
-                        self.idx = 0;
+                    match self.buf.try_extend_from_slice(ch) {
+                        Ok(_) => {}
+                        Err(_) => {
+                            warn!("Overfilled accumulator");
+                            self.buf.clear();
+                        }
                     }
```

Comment:
very minor, unimportant style nit: could just be 
```suggestion
                    if self.buf.try_extend_from_slice(ch).is_err() {
                         warn!("Overfilled accumulator");
                         self.buf.clear();
                    }
```